circuits/benches/valid_commitments.rs

```diff
@@ -67,35 +120,116 @@ pub fn bench_prover(c: &mut Criterion) {
 }
 
 /// Tests the time taken to verify `VALID COMMITMENTS`
-pub fn bench_verifier(c: &mut Criterion) {
-    // First generate a proof that will be verified multiple times
-    let (witness, statement) = create_default_witness_statement();
-    let mut transcript = HashChainTranscript::new(b"test");
-    let pc_gens = PedersenGens::default();
-    let prover = Prover::new(&pc_gens, &mut transcript);
+pub fn bench_verifier_with_sizes<
+    const MAX_BALANCES: usize,
+    const MAX_ORDERS: usize,
+    const MAX_FEES: usize,
+>(
+    c: &mut Criterion,
+) where
+    [(); MAX_BALANCES + MAX_ORDERS + MAX_FEES]: Sized,
+{
+    let mut group = c.benchmark_group("valid_commitments");
+    let benchmark_id = BenchmarkId::new(
+        "verifier",
+        format!("({MAX_BALANCES}, {MAX_ORDERS}, {MAX_FEES})"),
+    );
 
-    let (commitments, proof) = ValidCommitments::prove(witness, statement.clone(), prover).unwrap();
+    group.bench_function(benchmark_id, |b| {
+        // First generate a proof that will be verified multiple times
+        let (witness, statement) =
+            create_sized_witness_statement::<MAX_BALANCES, MAX_ORDERS, MAX_FEES>();
+        let mut transcript = HashChainTranscript::new(b"test");
+        let pc_gens = PedersenGens::default();
+        let prover = Prover::new(&pc_gens, &mut transcript);
+
+        let (commitments, proof) =
+            ValidCommitments::prove(witness, statement.clone(), prover).unwrap();
 
-    let mut group = c.benchmark_group("valid_commitments");
-    group.bench_function(BenchmarkId::new("verifier", ""), |b| {
         b.iter(|| {
             let mut transcript = HashChainTranscript::new(b"test");
             let verifier = Verifier::new(&pc_gens, &mut transcript);
 
-            ValidCommitments::verify(
+            let res = ValidCommitments::verify(
                 commitments.clone(),
                 statement.clone(),
                 proof.clone(),
                 verifier,
-            )
-            .unwrap();
+            );
+
+            #[allow(unused_must_use)]
+            {
+                black_box(res)
```

Comment:
Correct, we blackbox the output of the verification to avoid the compiler removing this.
1. Yes, in production we use the result so there is no optimization to be made. Blackboxing emulates a production build here.
2. No, in other benchmarks we use the result -- here we do not because `unwrap` will fail (it's annoying to construct a correct witness for benchmarking only)